react/__tests__/LocationInput.test.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@vtex/test-tools/react'
+import { MockedResponse } from '@apollo/react-testing'
+import { Address } from 'vtex.checkout-graphql'
+import { AddressRules, CountryRules, Field } from 'vtex.address-context/types'
+import { AddressContextProvider } from 'vtex.address-context/AddressContext'
+
+import LocationInput from '../LocationInput'
+import { validPostalCode } from '../__fixtures__/graphql/getAddressFromPostalCode.fixture'
+
+interface SetupProps {
+  address?: Address | null
+  countries?: string[]
+  rules?: AddressRules
+  graphqlMocks?: MockedResponse[]
+}
+
+interface RenderComponentProps extends SetupProps {
+  locationInputProps?: React.ComponentProps<typeof LocationInput>
+}
+
+const defaultGraphqlMocks: MockedResponse[] = []
+
+describe('LocationInput', () => {
+  const renderComponent = ({
+    locationInputProps = {},
+    address = { country: 'BRA' },
+    countries = [],
+    rules = {
+      BRA: {
+        fields: {
+          postalCode: {
+            mask: '99999-999',
+            pattern: '^(\\d){5}-?(\\d){3}$',
+            additionalData: {
+              forgottenURL:
+                'https://buscacepinter.correios.com.br/app/endereco/',
+            },
+          } as Field,
+        },
+      } as CountryRules,
+    },
+    graphqlMocks = [],
+  }: RenderComponentProps) => {
+    return render(
+      <AddressContextProvider
+        address={address}
+        countries={countries}
+        rules={rules}
+      >
+        <LocationInput {...locationInputProps} />
+      </AddressContextProvider>,
+      {
+        graphql: {
+          mocks: [...defaultGraphqlMocks, ...graphqlMocks],
+          addTypename: false,
+        },
+      }
+    )
+  }
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('should format and submit a valid postal code', async () => {
+    const mockedOnSuccess = jest.fn()
+
+    renderComponent({
+      locationInputProps: { onSuccess: mockedOnSuccess },
+      graphqlMocks: [validPostalCode],
+    })
+
+    const input = screen.getByRole('textbox', {
+      name: /postal code/i,
+    })
+
+    input.focus()
```

Comment:
why aren't you using `fireEvent` here? using the focus method on the dom element directly would require us to wrap this in `act` else we would get that act warning in the test run

this also applies to the other uses of the `focus` method in this test file

Reply:
weird, I didn't got act warnings, so I thought that it was ok. The reason why I'm not using `fireEvent` directly was due to the [link I've linked bellow](https://testing-library.com/docs/guide-events/#interactions-vs-events), but tbh I think this line in specific seems unecessary. Only the one in line 85 makes sense I guess